core/app/models/spree/variant.rb

```diff
@@ -385,6 +385,23 @@ def display_image(fallback: true)
       images.first || (fallback && product.variant_images.first) || Spree::Image.new
     end
 
+    # Images that can be used for the variant.
+    #
+    # Will first retrieve images on the variant. If it is an empty array,
+    # it'll fallback to master variant images.
+    # @return [Enumerable<Spree::Image>] images for the variant
+    def display_images
+      imgs = images
```

Comment:
Why are we creating a new variable in here? Can't we use `images` directly?

Reply:
Yeah, my bad.

Although looking it again, I found out that current code and the one you proposed does a `SELECT 1` + `SELECT` queries.

```
  Spree::Image Exists (0.3ms)  SELECT  1 AS one FROM "spree_assets" WHERE "spree_assets"."type" IN ('Spree::Image') AND "spree_assets"."viewable_id" = ? AND "spree_assets"."viewable_type" = ? LIMIT ?  [["viewable_id", 1], ["viewable_type", "Spree::Variant"], ["LIMIT", 1]]
  Spree::Image Load (0.3ms)  SELECT  "spree_assets".* FROM "spree_assets" WHERE "spree_assets"."type" IN ('Spree::Image') AND "spree_assets"."viewable_id" = ? AND "spree_assets"."viewable_type" = ? ORDER BY "spree_assets"."position" ASC LIMIT ?
```

Initially I proposed

```ruby
imgs = images
return imgs if imgs.size > 0
```
Which trigger a count query then return the collection proxy.

```
   (0.1ms)  SELECT COUNT(*) FROM "spree_assets" WHERE "spree_assets"."type" IN ('Spree::Image') AND "spree_assets"."viewable_id" = ? AND "spree_assets"."viewable_type" = ?  [["viewable_id", 1], ["viewable_type", "Spree::Variant"]]
  Spree::Image Load (0.2ms)  SELECT  "spree_assets".* FROM "spree_assets" WHERE "spree_assets"."type" IN ('Spree::Image') AND "spree_assets"."viewable_id" = ? AND "spree_assets"."viewable_type" = ? ORDER BY "spree_assets"."position" ASC LIMIT ?
```

I had to refactor because @houndci-bot complained about using `.size > 0`.

So, the best of both would be:

```ruby
return images if images.size > 0 # or return images if images.count > 0
```

WDYT?

Reply:
Perfect! :+1: